src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala

```diff
@@ -0,0 +1,102 @@
+package temple.generate.service.go
+
+import temple.generate.Endpoint
+import temple.generate.service.ServiceGenerator
+import temple.generate.utils.CodeTerm
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.utils.FileUtils
+import temple.utils.StringUtils.doubleQuote
+
+import scala.Option.when
+
+object GoServiceMainGenerator {
+
+  /** Given a service name, module name and whether the service uses inter-service communication, return the import
+    * block */
+  private[go] def generateImports(serviceName: String, module: String, usesComms: Boolean): String = {
+    val standardImports = Seq("flag", "log", "net/http").map(doubleQuote)
+
+    val customImports = Seq[CodeTerm](
+      when(usesComms) { s"${serviceName}Comm ${doubleQuote(s"$module/comm")}" },
+      s"${serviceName}DAO ${doubleQuote(s"$module/dao")}",
+      doubleQuote(s"$module/util"),
+      s"valid ${doubleQuote("github.com/asaskevich/govalidator")}",
+      doubleQuote("github.com/gorilla/mux"),
+    )
+
+    mkCode("import", CodeWrap.parens.tabbed(standardImports, "", customImports))
+  }
+
+  /** Given a service name and whether the service uses inter-service communication, return global statements */
+  private[go] def generateGlobals(serviceName: String, usesComms: Boolean): String =
+    mkCode.lines(
+      s"var dao ${serviceName}DAO.DAO",
+      when(usesComms) { s"var comm ${serviceName}Comm.Handler" },
+    )
+
+  /** Given a service name, whether the service uses inter-service communication, the endpoints desired and the port
+    * number, generate the main function */
+  private[go] def generateMain(serviceName: String, usesComms: Boolean, endpoints: Set[Endpoint], port: Int): String = {
+    val sb = new StringBuilder
+    sb.append("func main() ")
```

Comment:
This is unused, right?

Reply:
Yeah this should have been removed in your refactor PR, I'll remove it here.

Though in general I've tried to keep the order of code generating the same as the order of code, i.e. `"fun main() "` and then the body

Reply:
I guess we don't need the `body` variable, and we can maintain the order

Reply:
Yeah, this was my fault, sorry